web/impact/impact/tests/test_cancel_office_hour_reservation.py

```diff
@@ -36,7 +42,9 @@ def test_finalist_cancels_and_gets_ui_notification(self):
         response = self._submit_cancellation(office_hour,
                                              user=office_hour.finalist)
         notification = formatted_success_notification(office_hour)
-        self.assert_ui_notification(response, notification)
+        expected_ui_notification = get_ui_notification_dict(
```

Comment:
Can we maybe try to keep this clean? 

The consistent pattern here is, we have a success or failure, which conditions a success flag, a "header" and a "notification" - which are not actually header or notification, they are two fragments of a notification, which is being glommed together by the UI, but we'll use those words until we find the right ones. (which might not happen here, but should happen)

The "header" part and the "success" part are not independent, so we should only pass `success` and `notification`  to `self.assert_ui_notification` and let it decide what the "header" will be. Then we don't need a gratuitous `get_ui_notification_dict` function (which actually has nothing to do with notifications, and just composes a dict from the elements passed to it)

Also note that putting this into the APITestCase class implies that this pattern is going to be how we do UI notifications in the future, since we're building this concept into our tooling. I think this is a good thing and we should probably double down on it - for instance, it should probably be documented and maybe we should look for ways to incorporate this into our tooling on the views side as well as on the testing side.